sdk/identity/azure-identity/azure/identity/aio/_credentials/base.py

```diff
@@ -0,0 +1,28 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import abc
+
+
+class AsyncCredentialBase(abc.ABC):
+    @abc.abstractmethod
+    async def close(self):
+        pass
+
+    @abc.abstractmethod
+    async def __aenter__(self):
```

Comment:
Is there a reason we are not returning `self` here? Technically, returning `None` is allowed, but I'd rather follow `contextlib.AbstractAsyncContextManager`s lead (which does return `self`). Also, why is `__aenter__` an `@abc.abstractmethod` in the first place?

Reply:
Implementations all return `self`. `__aenter__` is abstract here because there isn't a good default. I suppose I could add an abstract `open` method and default to calling it, but I don't users to think they should deliberately "open" credentials.

Reply:
As I mentioned above, `contextlib` disagrees on the good default value - it returns `self`.